RELEASE_PROCEDURE.md

```diff
@@ -13,6 +13,9 @@ The release is prepared as follows:
 
 - Ask for a new libc version if, necessary. It usually is.  Then update the
   dependency in Cargo.toml accordingly.
+- Update the version number in `Cargo.toml`
```

Comment:
Added it to `RELEASE_PROCEDURE.md` based on my understanding of how our release procedure looks like, but I am actually not sure if this is correct